csvy/validators/registry.py

```diff
@@ -0,0 +1,38 @@
+"""Registry of validators to run on the header."""
+
+from collections.abc import Callable
+
+from pydantic import BaseModel
+
+VALIDATORS_REGISTRY: dict[str, type[BaseModel]] = {}
+"""Registry of validators to run on the header."""
+
+
+def register_validator(
+    name: str, overwrite: bool = False
+) -> Callable[[type[BaseModel]], type[BaseModel]]:
```

Comment:
I like the [`decorator`](https://pypi.org/project/decorator/) package for these situations. It also fixes up the type hints for decorated functions, which can otherwise be an issue.

I appreciate you may not want to add another dependency though!

Reply:
I've no problem with adding new dependencies, but I'm not convinced it adds much value in this case for just one, very simple decorator that just registers something and spits the same input.